### ISSUE TITLE
Use environment variable to set Android tools version when building

### ISSUE THREAD

**Issue by iambmelt** [AzureAD/azure-activedirectory-library-for-android#897]
Fortify SCA is currently configured to resolve `*.jar` dependencies from `android-build` using hardcoded paths. These paths include repetitive version numbers that are subject to change. Failure to update these constants means that changes to `buildToolsVersion` can mismatch VSTS-CI

```sh
\bin\sourceanalyzer.exe -b msalandroidbuild . 
-exclude ./**/androidTest/**/*.java 
-exclude ./**/test/**/*.java  
-exclude ./**/automationtestapp/**/R.java 
-exclude ./**/adal/**/R.java 
-exclude ./**/R.java 
-cp "C:\AndroidSDK\platforms\android-25\android.jar" 
-cp "C:\FortifyJarDependency\gson-2.2.4.jar" 
-cp "C:\FortifyJarDependency\android-support-customtabs.jar" 
-cp "C:\AndroidSDK\platforms\android-25\optional\org.apache.http.legacy.jar" 
-cp "C:\AndroidSDK\extras\android\m2repository\com\android\support\appcompat-v7\25.0.0\appcompat-v7-25.0.0-sources.jar" 
-cp "C:\AndroidSDK\extras\android\m2repository\com\android\support\support-fragment\25.0.0\support-fragment-25.0.0-sources.jar"
 -cp "C:\AndroidSDK\extras\android\m2repository\com\android\support\support-annotations\25.0.0\support-annotations-25.0.0-sources.jar" 
-jdk 1.8
```

**Comment by iambmelt** [comment]
The deliverable from this issue is successful deployment of the `Run Fortify` script in VSTS

**Comment by iambmelt** [comment]
Currently checking if Fortify can scan `*.aar` files -- seemingly no documented support but will try using `=ARCHIVE` flag. `-sources.jar` files do **not** work.

If `*.aar` files cannot be loaded, it may not be possible to resolve _all_ dependencies during Fortify scan, as the following classes are loaded from `*.aar` files
- `Nullable`
- `Fragment`
- `AppCompatActivity`
- `FragmentActivity`

**Comment by iambmelt** [comment]
Looks like the only way this is possible is to unzip the `*.aar`, and place them on the `<classpath>` using `-cp` -- not going to take this step at this time for the following reasons

- The `classes.jar` files contained in `*.aar` files are not meant to be consumed directly (sans resources)
- Consuming these `jars` directly creates additional work every time the `buildToolsVersion` or `compileSdkVersion` -- the associated `*.aar` would need to be copied, unzipped, moved, and then updated in the VSTS-CI script.

The only alternative to leaving these classes unresolved that I can think of would be to add another build phase where dependencies are copied from `$ANDROID_SDK_HOME`, passed into `unzip`, and renamed to give them predictable paths. Lots of work for little payoff.

**Comment by iambmelt** [comment]
Latest Fortify script takes the following form:

```sh
\bin\sourceanalyzer.exe -b msalandroidbuild .
-exclude ./**/androidTest/**/*.java
-exclude ./**/test/**/*.java
-exclude ./**/automationtestapp/**/R.java
-exclude ./**/adal/**/R.java
-exclude ./**/R.java
-exclude ./**/BuildConfig.java
-exclude ./**/IBrokerAccountService.java
-cp "C:\AndroidSDK\platforms\android-25\android.jar"
-cp "C:\FortifyJarDependency\gson-2.2.4.jar"
-cp "C:\FortifyJarDependency\android-support-customtabs.jar"
-cp "C:\AndroidSDK\platforms\android-25\optional\org.apache.http.legacy.jar"
-jdk 1.8
```
When `buildToolsVersion`, `targetSdkVersion`, `supportLibraryVersion`, and `compileSdkVersion` are advanced to `API 26` this script must be updated.

**Comment by iambmelt** [comment]
Added new build environment variable `android:apiVersion`

**Comment by iambmelt** [comment]
Added new build environment variable `android:gsonVersion`

**Comment by iambmelt** [comment]
Updated script to resolve correct jar through build environment variable. This change closes #897 and #914
```sh
\bin\sourceanalyzer.exe-b msalandroidbuild .
-exclude ./**/androidTest/**/*.java
-exclude ./**/test/**/*.java 
-exclude ./**/automationtestapp/**/R.java
-exclude ./**/adal/**/R.java
-exclude ./**/R.java
-exclude ./**/BuildConfig.java
-exclude ./**/IBrokerAccountService.java
-cp "C:\AndroidSDK\platforms\android-$(android:apiVersion)\android.jar"
-cp "C:\FortifyJarDependency\gson-$(android:gsonVersion).jar"
-cp "C:\FortifyJarDependency\android-support-customtabs.jar"
-cp "C:\AndroidSDK\platforms\android-$(android:apiVersion)\optional\org.apache.http.legacy.jar"
-jdk 1.8
```
